lib/ex_unit/lib/ex_unit/doc_test.ex

```diff
@@ -283,37 +289,47 @@ defmodule ExUnit.DocTest do
   defp gen_code_for_tests(tests, whole_expr, stack) do
     quote do
       stack = unquote(stack)
+
       try do
         # Put all tests into one context
-        unquote_splicing(tests)
+        (unquote_splicing(tests))
       rescue
         e in [ExUnit.AssertionError] ->
           reraise e, stack
 
         error ->
-          original_stack = System.stacktrace
+          original_stack = System.stacktrace()
+
           reraise ExUnit.AssertionError,
-            [message: "Doctest failed: got #{inspect(error.__struct__)} with message " <> inspect(Exception.message(error)),
-             expr: unquote(String.trim(whole_expr))],
-            original_stack
+                  [
+                    message: "Doctest failed: got #{inspect(error.__struct__)} with message " <>
+                      inspect(Exception.message(error)),
+                    expr: unquote(String.trim(whole_expr))
+                  ],
+                  original_stack
       end
     end
   end
 
   defp test_case_content(expr, {:test, expected}, location, stack) do
-    expr_ast     = string_to_quoted(location, stack, expr)
+    expr_ast = string_to_quoted(location, stack, expr)
     expected_ast = string_to_quoted(location, stack, expected)
 
     quote do
       expected = unquote(expected_ast)
+
       case unquote(expr_ast) do
-        ^expected -> :ok
+        ^expected ->
+          :ok
+
         actual ->
           reraise ExUnit.AssertionError,
-            [message: "Doctest failed",
-             expr: "#{unquote(String.trim(expr))} === #{unquote(String.trim(expected))}",
-             left: actual],
-            unquote(stack)
+                  [
+                    message: "Doctest failed",
```

Comment:
Same here, compute the message/expr or the whole keyword list on the previous line.

Reply:
you're the boss, but do you actually think it becomes more readable that way? IMHO, the keyword list is readable as it, and putting `message: message` anywhere doesn't help. But as I said, if you insist, I'll do the change, of course :)

Reply:
Yes, changing the message field here won't help but extracting the other fields out will. :)